ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1004,6 +1019,21 @@ void handleDataNodeConfigChange(Iterable<DataNodeConfig> configs, String dcName,
       }
     }
 
+    /**
+     * Handle deletion of an instance from cluster.
+     * @param instanceName instance that was deleted
+     */
+    void handleDataNodeDelete(String instanceName) {
+      synchronized (notificationLock) {
+        deleteInstanceInfo(instanceName);
```

Comment:
collapse deleteInstanceInfo here. So far i found only one usage of it and it looks like it must always be called from within this sync block. So don't open a path for someone to call it outside this sync block.

Reply:
Yeah, agree with you. Updated to inline the function.